sdk/src/network-client.ts

```diff
@@ -137,11 +137,12 @@ class AleoNetworkClient {
      * @param url The URL to fetch data from.
      */
     async fetchData<Type>(url = "/"): Promise<Type> {
-        try {
-            return parseJSON(await this.fetchRaw(url));
-        } catch (error) {
-            throw new Error(`Error fetching data: ${error}`);
-        }
+      try {
+        const raw = await retryWithBackoff(() => this.fetchRaw(url));
```

Comment:
The fetch here shouldn't have any retry since the retry is already taken care of in fetchRaw

```suggestion
        const raw = this.fetchRaw(url);
```